doc/authoring_command_modules/authoring_commands.md

```diff
@@ -77,108 +95,240 @@ with self.command_group('mymod', mymod_sdk) as g:
-- `completer` - The name of a callable that takes the following parameters `(prefix, action, parsed_args, **kwargs)` and return a list of completion values.
+While kwargs are inherited from higher levels on the diagram, they can be overriden at a lower level. For example, if `custom_command_type=foo` is used as a module-level kwarg in the `AzCommandLoader.__init__` method and `custom_command_type=bar` is passed for a call to `command_group`, then `bar` will be used for all calls to `custom_command` within that command group.
+
+Addtionally, you can see that kwargs registered on a command group *do not* carry over to argument contexts, so you must apply the kwargs in both places if necessary.
+
+****Commands Loader****
+
+_Special Kwargs_
+
+The following special kwargs are only interpretted by the command loader:
+- `min_profile` - Minimum profile which the module supports. If an older profile is used, the module will not be loaded.
+- `max_profile` - Maximum profile which the module supports. If a newer profile is used, the module will not be loaded.
+
+****Command Group****
+
+_Special Kwargs_
+
+The following special kwargs are supported by command group and its helper methods:
+- `table_transformer` - See section on [Custom Table Formats](#custom-table-formats)
+- `validator` - See section on [Validators](#validators)
+- `confirmation` - During interactive use, will prompt the user to confirm their choice to proceed. Supply a value of True to use the default prompt, or supply a string to use a custom prompt message. If the command is invoked in non-interactive scenarios and the --yes/-y parameter is not supplied, the command will fail.
+- `transform` - Accepts a callable that takes a command result, which can be manipulated as desired. The transformed result is then returned. In general, output formats should closely mirror those returned by the service, and so this should be infrequently used. The modifies the output *regardless of the output format type*.
+- `deprecate_info` - Accepts a string which will be displayed whenever the command is invoked. Used to display deprecation warnings.
+- `formatter_class` - Advanced. Accepts a custom class that derives from `argparse.HelpFormatter` to modify the help document generation.
+- `argument_loader` - Advanced. Accepts a callable that takes no parameters which will be used in place of the default argument loader.
+- `description_loader` - Advanced. Accepts a callable that takes no parameters which will be used in place of the default description loader.
+
+_General Kwargs_
+
+The following kwargs may be inherited from the command loader:
+- `min_api` - Minimum API version for which the command will appear.
+- `max_api` - Maximimum API version for which the command will appear.
+- `resource_type` - An `azure.cli.core.profiles.ResourceTYpe` enum value that is used for multi-API packages.
+- `operation_group` - Only used by the `azure-cli-vm` module to specify which resource API to target.
```

Comment:
Then is it worth adding it here if no-one else can use it?

Reply:
It is if (a) you're authoring a VM command and (b) if other SDKs go this route later on.